tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td

```diff
@@ -760,17 +760,17 @@ foreach ArgMinMaxOp = [TFL_ArgMinOp, TFL_ArgMaxOp] in {
      (AxesIsLastDimension $axes, $logits)]>;
 }
 
-class AllElementsAreI64<string val> : Constraint<CPred<
-  "($0.isa<DenseElementsAttr>() && "
-   "$0.cast<DenseElementsAttr>().getType().cast<ShapedType>().getElementType().isInteger(64) && "
-   "std::all_of($0.cast<DenseElementsAttr>().getValues<int64_t>().begin(), "
-               "$0.cast<DenseElementsAttr>().getValues<int64_t>().end(), "
-               "[](int64_t v){ return v == " #val# ";}))">>;
+class AllElementsAreI32OrI64<string val> : Constraint<CPred<
+  "($0.isa<DenseIntElementsAttr>() && "
+  "($0.cast<DenseIntElementsAttr>().getType().cast<ShapedType>().getElementType().isInteger(32) || "
+  "$0.cast<DenseIntElementsAttr>().getType().cast<ShapedType>().getElementType().isInteger(64)) && "
+  "llvm::all_of($0.cast<DenseIntElementsAttr>().getIntValues(), "
+               "[](APInt v){ return v.getSExtValue() == static_cast<int64_t>(" #val# ");}))">>;
 
 // Remove Slice ops slicing the whole input tensor, effectively no-op
 def OptimizeSliceOp : Pat<
   (TFL_SliceOp:$output $input, (ConstantOp $begin), $shape),
```

Comment:
To correct the above case, the shape argument should be checked to verify that it covers the original shape.

Reply:
Could you also add more test cases to cover the above case?

Reply:
Sure, I'll update it as soon as possible.

Reply:
It should work now. We are now checking shape of `input` with `shape`. This requires `shape` to be a constant. Let me know if it works smoothly :smiley: